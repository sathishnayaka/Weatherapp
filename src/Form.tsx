import React, {useState} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  TextInput,
  useColorScheme,
  View,
  Button,
} from 'react-native';

import {Colors} from 'react-native/Libraries/NewAppScreen';

type formTypes = {
  navigation: any;
};

function Form({navigation}: formTypes): JSX.Element {
  const [text, setText] = useState('');
  
  const onPressSubmit = () => {
    navigation.navigate('country-details', {
      country: text,
    });
  };

  const onChangeText = (val: string) => {
    setText(val);
  };

  return (
    <SafeAreaView >
      <View style={styles.sectionContainer}>
        <TextInput
          style={styles.input}
          onChangeText={onChangeText}
          value={text}
          placeholder="please enter country name"
          data-testid="text-input"
        />
        <Button
          title="submit"
          disabled={text === '' ? true : false}
          onPress={onPressSubmit}
          data-testid="submit-button"
        />
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
  input: {
    height: 40,
    margin: 12,
    borderWidth: 1,
    padding: 10,
  },
});

export default Form;
