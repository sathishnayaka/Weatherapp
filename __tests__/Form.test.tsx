import React from 'react';
import Form from '../src/Form';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
// jest.mock('axios');

describe('Form', () => {
  test('after the text-input it should navigate to country-details page with input text', () => {
    const navigation = { navigate: jest.fn() };
    const component = shallow(<Form navigation={navigation} />);
    expect(component).toMatchSnapshot();
    const textInput = component.find("[data-testid='text-input']");
    textInput.simulate('changeText', "india");
    const button = component.find("[data-testid='submit-button']");
    expect(button).toBeTruthy();
    button.simulate('press');
    expect(navigation.navigate).toHaveBeenCalled();
  });
});
