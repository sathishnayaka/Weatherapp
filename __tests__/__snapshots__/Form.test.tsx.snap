// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form after the text-input it should navigate to country-details page with input text 1`] = `ShallowWrapper {}`;
